### ISSUE TITLE
Update Emby Rock-on

### ISSUE THREAD

**Issue by FroggyFlox** [rockstor/rockon-registry#152]
Emby server recently released all its versions on DotNet and thus changed its docker image substantially. While the current Json Rock-on still works, updating it would bring a few new features.
https://hub.docker.com/r/emby/embyserver/

I'll try to spend time on this and am creating this issue to remind me and keep track it it.

**Comment by FroggyFlox** [comment]
Hi @phillxnet ,

How can one test a new version of a Rock-on that already is on the official Repository?
When trying to load a differentially named JSON loaded in `/opt/rockstor/rockons-metastore/` I get the following error:
```
Houston, we've had a problem.
Errors occurred while processing updates for following Rock-ons. EmbyServerDotNET: ['Duplicate container(embyserver) definition detected. It belongs to another Rock-on(EmbyServer). Uninstall one of them and try again.', 'Traceback (most recent call last):\n File "/opt/rockstor/eggs/gunicorn-19.7.1-py2.7.egg/gunicorn/workers/sync.py", line 68, in run_for_one\n self.accept(listener)\n File "/opt/rockstor/eggs/gunicorn-19.7.1-py2.7.egg/gunicorn/workers/sync.py", line 27, in accept\n client, addr = listener.accept()\n File "/usr/lib64/python2.7/socket.py", line 202, in accept\n sock, addr = self._sock.accept()\nerror: [Errno 11] Resource temporarily unavailable\n']
```
```
      Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/storageadmin/views/rockon.py", line 106, in post
    self._create_update_meta(r, rockons[r])
  File "/opt/rockstor/eggs/Django-1.8.16-py2.7.egg/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/opt/rockstor/src/rockstor/storageadmin/views/rockon.py", line 185, in _create_update_meta
    handle_exception(Exception(e_msg), self.request)
  File "/opt/rockstor/src/rockstor/storageadmin/util.py", line 47, in handle_exception
    trace=traceback.format_exc())
RockStorAPIException: ['Duplicate container(embyserver) definition detected. It belongs to another Rock-on(EmbyServer). Uninstall one of them and try again.', 'Traceback (most recent call last):\n  File "/opt/rockstor/eggs/gunicorn-19.7.1-py2.7.egg/gunicorn/workers/sync.py", line 68, in run_for_one\n    self.accept(listener)\n  File "/opt/rockstor/eggs/gunicorn-19.7.1-py2.7.egg/gunicorn/workers/sync.py", line 27, in accept\n    client, addr = listener.accept()\n  File "/usr/lib64/python2.7/socket.py", line 202, in accept\n    sock, addr = self._sock.accept()\nerror: [Errno 11] Resource temporarily unavailable\n']

   
```

**Comment by schakrava** [comment]
@FroggyFlox Can you name the embyserver container differently? That should fix the problem.

**Comment by FroggyFlox** [comment]
Unfortunately the container is the same so I can't name it differently.
One option is for me to fork the container for testing purposes only and use that to test the updated JSON. Once it's tested and verified to work I could then change the container back to the actual `emby/embyserver` and submit that for PR.

Alternatively, is there a way to change the location from where the Rock-ons are fetched on my test system? I could thus try to target the corresponding branch of my rockon-registry fork to test it without needing to fork the container itself.

**Comment by FroggyFlox** [comment]
The main (and only important?) new feature would be the support of hardware transcoding, which requires adding the support for the `--device` mapping in the JSON file.
However, as this would be needed (and possible) only for users with components compatible with hardware transcoding, this would need to be implemented as an additional option during RockOn install.
Follow the same implementation as for the "options" but with user input.

**Comment by FroggyFlox** [comment]
For those interested, a corresponding PR was submitted:
https://github.com/rockstor/rockstor-core/pull/1962

**Comment by wiza** [comment]
Any plan on merging this?

**Comment by wiza** [comment]
I still can't see device options after clicking update and running install?

**Comment by phillxnet** [comment]
@wiza The changes by @FroggyFlox against this issue have been committed to the GitHub repo, we are just waiting on it's contents being published so that Rockstor installs can then pick up on those changes.
Hopefully this shouldn't be too much longer as we now have quite a few changes awaiting the same 'publish' step.
Thanks for you patience.

**Comment by FroggyFlox** [comment]
> I still can't see device options after clicking update and running install?

Hi @wiza,

This is likely due to the fact that even though the changes were merged into the PR, they still need to be pushed into the actual file from which Rockstor fetches Rock-ons. I'm unfortunately not familiar with that process so this is something on which @phillxnet and @schakrava should be able to provide more information. 
I'm sorry for the trouble.

**Comment by wiza** [comment]
That’s ok, but how do I know when it’s actually updated (without unnecessary deletion and install)? Are rockons somewhere locally or is the server where those are fetched open for public browsing?
